### PR TITLE
[Transform] correctly retrieve checkpoints from remote indices

### PR DESCRIPTION
**Pull request by hendrikmuhs**
uses remote client(s) to correctly retrieve index checkpoints from remote clusters

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/RemoteClusterResolver.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        public Map<String, List<String>> getRemoteIndicesPerClusterAlias() {
+            return remoteIndicesPerClusterAlias;
+        }
+
+        public List<String> getLocalIndices() {
+            return localIndices;
+        }
+
+        public int size() {
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/RemoteClusterResolver.java]:
```suggestion
        public int numClusters() {
```

Or something. I thought `size()` was meant to be the number of total resolved indices, but instead it is used to help allocate the grouped listener.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java [hunk 2]
```diff
@@ -84,13 +93,61 @@ public void createNextCheckpoint(final TransformCheckpoint lastCheckpoint, final
...
+
+                groupedListener = new GroupedActionListener<>(mergeMapsListener, resolvedIndexes.size());
+            }
+
+            if (resolvedIndexes.getLocalIndices().isEmpty() == false) {
+                getCheckpointsFromOneCluster(
+                    client,
+                    transformConfig.getHeaders(),
+                    resolvedIndexes.getLocalIndices().toArray(new String[0]),
+                    "",
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java]:
```suggestion
                    LOCAL_CLUSTER_GROUP_KEY,
```

**Review by benwtrent — APPROVED** [review]

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc 
run elasticsearch-ci/default-distro

**Pull request merged**